api/v1alpha1/circuitrelay_types.go

```diff
@@ -0,0 +1,104 @@
+/*
+Copyright 2022.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	"github.com/libp2p/go-libp2p-core/peer"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	ma "github.com/multiformats/go-multiaddr"
+)
+
+// This is intended to mimic peer.AddrInfo
+type AddrInfoBasicType struct {
+	ID       string        `json:"id"`
+	Addrs    []string      `json:"addrs"`
+	addrInfo peer.AddrInfo `json:"-"`
```

Comment:
Since this struct is mimicking `peer.AddrInfo`, do we need this `addrInfo` field here, or could we get rid of this and simply treat `AddrInfoBasicType` as if it's `peer.AddrInfo`?

Reply:
Really what I'm trying to do is serialize the peer.AddrInfo so we can use it in methods that want it. This peer.AddrInfo struct does not have json tags, and it can't be deserialized using kubernetes' json decoder.

Reply:
Okay, we can leave it in for now